File: tests/unit/java/com/trulioo/sdk/model/VerifyRequestTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import java.util.List;
import java.util.Arrays;

public class VerifyRequestTest {
    private VerifyRequest verifyRequest;

    @BeforeEach
    public void beforeEach() {
        verifyRequest = new VerifyRequest();
    }

    @Test
    public void testAcceptTruliooTermsAndConditions() {
        Boolean acceptTruliooTermsAndConditions = true;
        verifyRequest.setAcceptTruliooTermsAndConditions(acceptTruliooTermsAndConditions);

        assertEquals(acceptTruliooTermsAndConditions, verifyRequest.getAcceptTruliooTermsAndConditions());
        assertEquals(acceptTruliooTermsAndConditions, new VerifyRequest()
                .acceptTruliooTermsAndConditions(acceptTruliooTermsAndConditions).getAcceptTruliooTermsAndConditions());
    }

    @Test
    public void testDemo() {
        Boolean demo = false;
        verifyRequest.setDemo(demo);

        assertEquals(demo, verifyRequest.getDemo());
        assertEquals(demo, new VerifyRequest().demo(demo).getDemo());
    }

    @Test
    public void testCallBackUrl() {
        String callBackUrl = "https://trulioo.com";
        verifyRequest.setCallBackUrl(callBackUrl);

        assertEquals(callBackUrl, verifyRequest.getCallBackUrl());
        assertEquals(callBackUrl, new VerifyRequest().callBackUrl(callBackUrl).getCallBackUrl());
    }

    @Test
    public void testTimeout() {
        Integer timeout = 3600;
        verifyRequest.setTimeout(timeout);

        assertEquals(timeout, verifyRequest.getTimeout());
        assertEquals(timeout, new VerifyRequest().timeout(timeout).getTimeout());
    }

    @Test
    public void testCleansedAddress() {
        Boolean cleansedAddress = false;
        verifyRequest.setCleansedAddress(cleansedAddress);

        assertEquals(cleansedAddress, verifyRequest.getCleansedAddress());
        assertEquals(cleansedAddress, new VerifyRequest().cleansedAddress(cleansedAddress).getCleansedAddress());
    }

    @Test
    public void testConfigurationName() {
        String configurationName = "Identity Verification";
        verifyRequest.setConfigurationName(configurationName);

        assertEquals(configurationName, verifyRequest.getConfigurationName());
        assertEquals(configurationName,
                new VerifyRequest().configurationName(configurationName).getConfigurationName());
    }

    @Test
    public void testConsentForDataSources() {
        verifyRequest.addConsentForDataSourcesItem("no");
        verifyRequest.addConsentForDataSourcesItem("no");
        List<String> consentForDataSources = Arrays.asList("yes");
        verifyRequest.setConsentForDataSources(consentForDataSources);

        assertEquals(consentForDataSources, verifyRequest.getConsentForDataSources());
        assertEquals(consentForDataSources,
                new VerifyRequest().consentForDataSources(consentForDataSources).getConsentForDataSources());
    }

    @Test
    public void testCountryCode() {
        String countryCode = "CA";
        verifyRequest.setCountryCode(countryCode);

        assertEquals(countryCode, verifyRequest.getCountryCode());
        assertEquals(countryCode, new VerifyRequest().countryCode(countryCode).getCountryCode());
    }

    @Test
    public void testCustomerReferenceID() {
        String customerReferenceID = "ref-123";
        verifyRequest.setCustomerReferenceID(customerReferenceID);

        assertEquals(customerReferenceID, verifyRequest.getCustomerReferenceID());
        assertEquals(customerReferenceID,
                new VerifyRequest().customerReferenceID(customerReferenceID).getCustomerReferenceID());
    }

    @Test
    public void testDataFields() {
        DataFields dataFields = new DataFields().personInfo(new PersonInfo().firstGivenName("A"));
        verifyRequest.setDataFields(dataFields);

        assertEquals(dataFields, verifyRequest.getDataFields());
        assertEquals(dataFields, new VerifyRequest().dataFields(dataFields).getDataFields());
    }

    @Test
    public void testVerboseMode() {
        Boolean verboseMode = false;
        verifyRequest.setVerboseMode(verboseMode);

        assertEquals(verboseMode, verifyRequest.getVerboseMode());
        assertEquals(verboseMode, new VerifyRequest().verboseMode(verboseMode).getVerboseMode());
    }

    @Test
    public void testEquals() {
        Boolean acceptTruliooTermsAndConditions = true;
        VerifyRequest verifyRequest1 = new VerifyRequest()
                .acceptTruliooTermsAndConditions(acceptTruliooTermsAndConditions);

        assertEquals(verifyRequest1, verifyRequest1);
        assertEquals(verifyRequest1,
                new VerifyRequest().acceptTruliooTermsAndConditions(acceptTruliooTermsAndConditions));
        assertNotEquals(verifyRequest1,
                new VerifyRequest().acceptTruliooTermsAndConditions(!acceptTruliooTermsAndConditions));
        assertNotEquals(verifyRequest1, null);
    }

    @Test
    public void testHashcode() {
        Boolean acceptTruliooTermsAndConditions = true;

        assertEquals(new VerifyRequest().acceptTruliooTermsAndConditions(acceptTruliooTermsAndConditions).hashCode(),
                new VerifyRequest().acceptTruliooTermsAndConditions(acceptTruliooTermsAndConditions).hashCode());
    }

    @Test
    public void testToString() {
        Boolean acceptTruliooTermsAndConditions = true;

        assertEquals(new VerifyRequest().acceptTruliooTermsAndConditions(acceptTruliooTermsAndConditions).toString(),
                new VerifyRequest().acceptTruliooTermsAndConditions(acceptTruliooTermsAndConditions).toString());
    }
}

File: tests/unit/php/Model/VerifyRequestTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use InvalidArgumentException;
use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\VerifyRequest;
use Trulioo\SDK\Model\DataFields;
use Trulioo\SDK\Model\PersonInfo;

class VerifyRequestTest extends TestCase
{
    private VerifyRequest $verifyRequest;

    public function setUp(): void
    {
        $this->verifyRequest = new VerifyRequest([]);
    }

    public function testPropertyAcceptTruliooTermsAndConditions()
    {
        $this->verifyRequest->setAcceptTruliooTermsAndConditions(true);
        $this->assertEquals(true, $this->verifyRequest->getAcceptTruliooTermsAndConditions());
    }

    public function testPropertyDemo()
    {
        $this->verifyRequest->setDemo(false);
        $this->assertEquals(false, $this->verifyRequest->getDemo());
    }

    public function testPropertyCallBackUrl()
    {
        $this->verifyRequest->setCallBackUrl('https://trulioo.com');
        $this->assertEquals('https://trulioo.com', $this->verifyRequest->getCallBackUrl());

        try {
            $this->verifyRequest->setCallBackUrl(str_repeat('a', 2084));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyTimeout()
    {
        $this->verifyRequest->setTimeout(3600);
        $this->assertEquals(3600, $this->verifyRequest->getTimeout());
    }

    public function testPropertyCleansedAddress()
    {
        $this->verifyRequest->setCleansedAddress(true);
        $this->assertEquals(true, $this->verifyRequest->getCleansedAddress());
    }

    public function testPropertyConfigurationName()
    {
        $this->verifyRequest->setConfigurationName('IDV');
        $this->assertEquals('IDV', $this->verifyRequest->getConfigurationName());

        try {
            $this->verifyRequest->setConfigurationName(str_repeat('a', 46));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyConsentForDataSources()
    {
        $this->verifyRequest->setConsentForDataSources(['yes']);
        $this->assertEquals(
            ['yes'],
            $this->verifyRequest->getConsentForDataSources()
        );
    }

    public function testPropertyCountryCode()
    {
        $this->verifyRequest->setCountryCode('AU');
        $this->assertEquals('AU', $this->verifyRequest->getCountryCode());
    }

    public function testPropertyCustomerReferenceId()
    {
        $this->verifyRequest->setCustomerReferenceId('ref-123');
        $this->assertEquals('ref-123', $this->verifyRequest->getCustomerReferenceId());

        try {
            $this->verifyRequest->setCustomerReferenceId(str_repeat('a', 129));
            $this->fail('Expected exception');
        } catch (InvalidArgumentException $e) {
        }
    }

    public function testPropertyDataFields()
    {
        $this->verifyRequest->setDataFields(new DataFields([
            new PersonInfo([
                'first_given_name' => 'A'
            ])
        ]));
        $this->assertEquals(
            new DataFields([
                new PersonInfo([
                    'first_given_name' => 'A'
                ])
            ]),
            $this->verifyRequest->getDataFields()
        );
    }

    public function testPropertyVerboseMode()
    {
        $this->verifyRequest->setVerboseMode(true);
        $this->assertEquals(true, $this->verifyRequest->getVerboseMode());
    }
}
